client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/backend/issue/IssueService.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SonarLint Core - Client API
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.clientapi.backend.issue;
+
+import java.util.concurrent.CompletableFuture;
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
+
+public interface IssueService {
+
+  @JsonRequest
+  CompletableFuture<Void> changeStatus(ChangeIssueStatusParams params);
+
+  /**
+   * Adds a new comment to an issue (specified by {@link AddIssueCommentParams#getIssueKey()})
+   */
+  @JsonRequest
+  CompletableFuture<Void> addComment(AddIssueCommentParams params);
```

Comment:
Lately we are also trying to better document services' methods, with the expected functional behavior and edge cases. You could have a look at `HotspotService`

Reply:
Addressed in [e3eedf4](https://github.com/SonarSource/sonarlint-core/pull/612/commits/e3eedf4a8d30bebab82f68b85844e20e2314b3f0)